### ISSUE TITLE
Question: Why is MemoryFileUpload is not released by DefaultHttpDataFactory?

### ISSUE THREAD

**Issue by jameskleeh** [netty/netty#8640]
### Expected behavior
I expect any MemoryFileUploads to be released when any of the cleanAll methods are called

### Actual behavior
Mixed and Disk uploads are cleaned, but memory isn't. This seems intentional as the same logic is applied to attributes.

The upload/attributes are never added to the list that gets iterated over when cleaning

https://github.com/netty/netty/blob/4.1/codec-http/src/main/java/io/netty/handler/codec/http/multipart/DefaultHttpDataFactory.java#L245

### Netty version
4.1.30

### JVM version (e.g. `java -version`)
Java 8

### OS version (e.g. `uname -a`)
macOS

**Comment by jameskleeh** [comment]
@normanmaurer Sure thing

**Comment by jameskleeh** [comment]
@normanmaurer Done https://github.com/netty/netty/pull/9672

**Comment by fredericBregier** [comment]
@jameskleeh Indeed, that was intentional. Maybe wrongly, since you feel it should be included...
Your PR seems OK to change the behavior, but could you also fix the comments on the related "clean" methods, such that it does not explicitely stands for "disk" only (or mixed so mixed too) HttpData ?

**Comment by jameskleeh** [comment]
@fredericBregier Sure thing

**Comment by jameskleeh** [comment]
@fredericBregier I don't see what comments you are referring to. Can you point me in the right direction?

**Comment by fredericBregier** [comment]
Sure:

https://github.com/netty/netty/blob/c70254e3af242c6134063bbb3d6f4ade8775a4ff/codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpDataFactory.java#L75

https://github.com/netty/netty/blob/c70254e3af242c6134063bbb3d6f4ade8775a4ff/codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpDataFactory.java#L80

https://github.com/netty/netty/blob/c70254e3af242c6134063bbb3d6f4ade8775a4ff/codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpData.java#L121
(even if this one was already wrong, since in AbstractMemoryHttpData, the delete method indeed call release on bytebuf)

I don't go everywhere, but I think this are the main ones.

Of course, we might also add a new method `clean` duplicate of :
https://github.com/netty/netty/blob/c70254e3af242c6134063bbb3d6f4ade8775a4ff/codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java#L660
since it now will delete all, even not files, in order to be clean (even if cleanFile will delete all). Or just changing the associated comment to ensure the new behavior is understood.
And same on https://github.com/netty/netty/blob/c70254e3af242c6134063bbb3d6f4ade8775a4ff/codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java#L949
